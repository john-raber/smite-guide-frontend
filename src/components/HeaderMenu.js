import React, { Fragment } from "react";
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from "reactstrap";
import { FaSignInAlt, FaSignOutAlt } from "react-icons/fa";

const HeaderMenu = () => {
  return (
    <Fragment>
      <Navbar dark fixed>
        <NavbarBrand>SmiteGuide</NavbarBrand>
        <Nav>
          <NavItem>
            <NavLink>About</NavLink>
          </NavItem>
          <NavItem>
            <NavLink>Gods</NavLink>
          </NavItem>
          <NavItem>
            <NavLink>Items</NavLink>
          </NavItem>
          <NavItem>
            <NavLink>Game Modes</NavLink>
          </NavItem>
          <NavItem>
            <NavLink>Patch Notes</NavLink>
          </NavItem>
        </Nav>
        <Nav>
          <NavItem>
            <NavLink>
              <FaSignInAlt />
            </NavLink>
          </NavItem>
        </Nav>
      </Navbar>
    </Fragment>
  );
};

export default HeaderMenu;
