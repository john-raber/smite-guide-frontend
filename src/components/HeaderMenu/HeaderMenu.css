/* Setting the color and display type of 'SmiteGuide' in the HeaderMenu */
.navbar-brand p {
  display: inline;
  color: rgb(255, 255, 255);
}

/* Setting the color of links in the HeaderMenu */
.nav-item a.nav-link {
  color: rgb(255, 255, 255);
}
