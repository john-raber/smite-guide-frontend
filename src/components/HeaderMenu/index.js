import React, { Fragment } from 'react';
import { connect } from 'react-redux';
import { Navbar, NavbarBrand, Nav, NavItem, NavLink } from 'reactstrap';
import { FaBolt } from 'react-icons/fa';

import './HeaderMenu.css';
import CurrentUserOptions from './components/CurrentUserOptions/index';

const HeaderMenu = () => {
  return (
    <Fragment>
      <Navbar className="headerMenu">
        <NavbarBrand href="/">
          <FaBolt color="#ecc345" />
          <p> SmiteGuide</p>
        </NavbarBrand>
        <Nav>
          <NavItem>
            <NavLink href="/about">About</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="/gods">Gods</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="/items">Items</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="/game-modes">Game Modes</NavLink>
          </NavItem>
          <NavItem>
            <NavLink href="/patch-notes">Patch Notes</NavLink>
          </NavItem>
        </Nav>
        <Nav>
          <NavItem>
            <CurrentUserOptions />
          </NavItem>
        </Nav>
      </Navbar>
    </Fragment>
  );
};

const mapStateToProps = state => {
  return {
    currentUser: state.currentUser
  };
};

export default connect(mapStateToProps)(HeaderMenu);
export { HeaderMenu };
