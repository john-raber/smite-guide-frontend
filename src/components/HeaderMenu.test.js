import React from "react";
import { shallow } from "enzyme";

import { HeaderMenu } from "./HeaderMenu/index";

describe("HeaderMenu", () => {
  it("renders correctly", () => {
    let wrapper = shallow(<HeaderMenu />);

    expect(wrapper.find(".headerMenu").length).toEqual(1);
  });

  it("matches its snapshot when a currentUser is set", () => {
    let user = { email: "john@example.com" };
    let wrapper = shallow(<HeaderMenu currentUser={user} />);

    expect(wrapper).toMatchSnapshot();
  });

  it("matches its snapshot when no currentUser is set", () => {
    let wrapper = shallow(<HeaderMenu />);

    expect(wrapper).toMatchSnapshot();
  });
});
