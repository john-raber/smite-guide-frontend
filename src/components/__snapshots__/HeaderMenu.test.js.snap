// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeaderMenu matches its snapshot when a currentUser is set 1`] = `
<Fragment>
  <Navbar
    className="headerMenu"
    expand={false}
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <FaBolt
        color="#ecc345"
      />
       
      <p>
        SmiteGuide
      </p>
    </NavbarBrand>
    <Nav
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          href="/about"
          tag="a"
        >
          About
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/gods"
          tag="a"
        >
          Gods
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/items"
          tag="a"
        >
          Items
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/game-modes"
          tag="a"
        >
          Game Modes
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/patch-notes"
          tag="a"
        >
          Patch Notes
        </NavLink>
      </NavItem>
    </Nav>
    <Nav
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <Connect(CurrentUserOptions) />
      </NavItem>
    </Nav>
  </Navbar>
</Fragment>
`;

exports[`HeaderMenu matches its snapshot when no currentUser is set 1`] = `
<Fragment>
  <Navbar
    className="headerMenu"
    expand={false}
    tag="nav"
  >
    <NavbarBrand
      href="/"
      tag="a"
    >
      <FaBolt
        color="#ecc345"
      />
       
      <p>
        SmiteGuide
      </p>
    </NavbarBrand>
    <Nav
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <NavLink
          href="/about"
          tag="a"
        >
          About
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/gods"
          tag="a"
        >
          Gods
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/items"
          tag="a"
        >
          Items
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/game-modes"
          tag="a"
        >
          Game Modes
        </NavLink>
      </NavItem>
      <NavItem
        tag="li"
      >
        <NavLink
          href="/patch-notes"
          tag="a"
        >
          Patch Notes
        </NavLink>
      </NavItem>
    </Nav>
    <Nav
      tag="ul"
      vertical={false}
    >
      <NavItem
        tag="li"
      >
        <Connect(CurrentUserOptions) />
      </NavItem>
    </Nav>
  </Navbar>
</Fragment>
`;
